src/formatDate.js

```diff
@@ -9,6 +9,48 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  let newFormatedDate = '';
+  const oldSeparator = fromFormat[3];
+  const splitedOldDate = date.split(oldSeparator);
+  const dateMap = getDate(fromFormat, splitedOldDate);
+  const newSeparator = toFormat[3];
+
+  for (let i = 0; i < 3; i++) {
+    if (newFormatedDate.length) {
+      newFormatedDate += newSeparator;
+    }
+
+    newFormatedDate += dateMap[toFormat[i]];
+  }
+
+  return newFormatedDate;
```

Comment:
```suggestion
 newFormatedDate = toFormat.reduce((acc, formatPart) => [...acc, dateMap[formatPart]], []).join(newSeparator)
  }
```

Have you tried doing this?

Reply:
I tried, but the input array has a length of 4 elements, so an extra separator was added at the end, after sending the task, I realized that this could be solved by assigning the 'separator' variable a value not to a straight line, but using the 'pop' method